shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -111,6 +112,23 @@ public InputConnectionAdaptor(
     mImm = (InputMethodManager) view.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
   }
 
+  interface GetOffsetBefore {
+    int getOffsetBefore(String text, int offset);
+  }
+
+  private GetOffsetBefore getOffsetBefore =
+      new GetOffsetBefore() {
+        @Override
+        public int getOffsetBefore(String text, int offset) {
+          return FlutterJNI.nativeGetTextOffsetBefore(text, offset);
+        }
+      };
```

Comment:
nit: Don't we have access to lambdas?

Reply:
AFAIK the Android embedding does not use lambdas.  The embedding supports Android versions back to API 16 and tries to avoid anything that would make it more difficult to support legacy toolchains and runtimes.